app/src/main/java/msi/paria/trainingapp/pages/main/MainActivity.kt

```diff
@@ -0,0 +1,44 @@
+package msi.paria.trainingapp.pages.main
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.lifecycle.ViewModelProvider
+import msi.paria.trainingapp.R
+import msi.paria.trainingapp.pages.main.view.RegisterPageView
+import msi.paria.trainingapp.pages.main.view_model.RegisterViewModel
+
+
+class MainActivity : ComponentActivity() {
+
+    private lateinit var registerViewModel: RegisterViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        registerViewModel = ViewModelProvider(this)[RegisterViewModel::class.java]
```

Comment:
Use a built-in extention function for this. 
[See this link](https://developer.android.com/topic/libraries/architecture/viewmodel/viewmodel-apis#:~:text=The%20viewModels()%20extension%20functions,scoped%20to%20the%20closest%20ViewModelStoreOwner%20.)